internal/schedule/scheduler_test.go

```diff
@@ -0,0 +1,37 @@
+package schedule
+
+import (
+	"context"
+	"testing"
+	"time"
+)
+
+func Test_CancellingContextShouldStopTask(t *testing.T) {
+	scheduler := NewScheduler()
+	timeout, cancel := context.WithCancel(context.Background())
+
+	cancelContext := time.NewTimer(1001 * time.Microsecond)
```

Comment:
btw if timer is set to ~1,5ms task is executed 3 times, not 2
but I guess it is impossible to guarantee sub-millisecond synchronization with channels?